src/javascript/editorTabs/EditPanelContent/Preview/PreviewViewers/IframeViewer/IframeViewer.jsx

```diff
@@ -49,61 +49,50 @@ function loadAssets(assets, iframeDocument) {
     return Promise.all(assets.map(asset => loadAsset(asset, iframeHeadEl)));
 }
 
-function writeInIframe(html, iframeWindow) {
-    return new Promise((resolve, reject) => {
-        iframeWindow.document.open();
-        iframeWindow.onload = resolve;
-        iframeWindow.onerror = reject;
-        iframeWindow.document.write(html);
-        iframeWindow.document.close();
-    });
-}
-
 export const IframeViewer = ({previewContext, data, onContentNotFound}) => {
     const [loading, setLoading] = useState(true);
     const editorContext = useContentEditorContext();
     const {t} = useTranslation('content-editor');
     const iframeRef = useRef(null);
+    const onLoadTimeoutRef = useRef(null);
+    let displayValue = data && data.nodeByPath && data.nodeByPath.renderedContent ? data.nodeByPath.renderedContent.output : '';
+    if (displayValue === '') {
+        displayValue = t('label.contentManager.contentPreview.noViewAvailable');
+    }
 
-    const renderIFrame = useCallback(() => {
-        const element = iframeRef.current;
-
-        if (!element) {
-            return;
-        }
-
-        setLoading(true);
-        let displayValue = data && data.nodeByPath && data.nodeByPath.renderedContent ? data.nodeByPath.renderedContent.output : '';
-        if (displayValue === '') {
-            displayValue = t('label.contentManager.contentPreview.noViewAvailable');
+    useEffect(() => {
+        // Add a timer to always remove the loader and click on links after 5 seconds
+        onLoadTimeoutRef.current = setTimeout(() => {
+            console.debug('iframe onLoad did not trigger, remove loader and disable click events');
+            const element = iframeRef.current;
+            const iframeWindow = element.contentWindow || element;
+            iframeWindow.document.body.setAttribute('style', 'pointer-events: none');
```

Comment:
Since the timeout was trigger, we cannot know the state of the iframe.
Do you think we should add undefined protection on this ?
- element ?
- iframeRef.current ?
- iframeWindow.document ? .body ?